client/blocks/shortcode/index.js

```diff
@@ -69,7 +69,7 @@ Shortcode.propTypes = {
 };
 
 Shortcode.defaultProps = {
-	filterRenderResult: identity,
+	filterRenderResult: ( result ) => result,
```

Comment:
It should be straighforward to make the (one) `filterRenderResult` call in this component only when the prop is defined -- and get rid of the default prop.

Reply:
Good call. Updated in https://github.com/Automattic/wp-calypso/pull/52383/commits/bc6e877885391b27ff2b93140908d33f211f8ab3.